### PR TITLE
Implement React Query, loading and error states

### LINKED ISSUE #58: Implement React Query, loading and error states
- React Query for more efficient fetching
- Implement loading and error states with animation/info on all relevant occasions

### PR DESCRIPTION
**Pull request by ncbecker**
Closes #58

### CODE REVIEW

**Review by mhstrkmp — APPROVED** [review]
Nice and clean code lgtm 🚀

File: client/src/pages/EpisodeDetailsPage.js [hunk 1]
```diff
@@ -116,21 +117,21 @@ const NotLikedSmall = styled(NotLiked)`
...
-    const doFetch = async () => {
-      const details = await getEpisodeDetailsFromDB(id, user.id);
-      setEpisodeDetails(details);
-    };
-    if (id && user.id) {
-      doFetch();
+  const { data: episodeDetails, status: statusEpisodeDetails } = useQuery(
+    ["voting", id, user.id],
+    async () => {
+      if (id && user.id) {
```
Comment (bwnstck) [inline client/src/pages/EpisodeDetailsPage.js:130]: Ternary 4tw

File: client/src/pages/VotingPage.js [hunk 2]
```diff
@@ -61,41 +62,58 @@ const CardsWrapper = styled.div`
...
   const theme = useTheme().theme;
 
+  const {
+    data: votedEpisodes,
+    status: statusVotedEpisodes,
+    refetch: refetchVotedEpisodes,
+  } = useQuery(
+    ["voting", user],
+    async () => {
+      if (user) {
```
Comment (bwnstck) [inline client/src/pages/VotingPage.js]: Same here :)

**Review by bwnstck — APPROVED** [review]
Badabam

**Pull request merged**